src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidatorProvider.cs

```diff
@@ -16,11 +18,42 @@ namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
     /// </summary>
     public class DataAnnotationsModelValidatorProvider : IModelValidatorProvider
     {
+        private IOptions<MvcDataAnnotationsLocalizationOptions> _options;
+        private IStringLocalizerFactory _stringLocalizerFactory;
+
+        public DataAnnotationsModelValidatorProvider(
+            IOptions<MvcDataAnnotationsLocalizationOptions> options,
+            IStringLocalizerFactory stringLocalizerFactory)
+        {
+            _options = options;
+            _stringLocalizerFactory = stringLocalizerFactory;
+        }
+
         public void GetValidators(ModelValidatorProviderContext context)
         {
+            IStringLocalizer stringLocalizer = null;
+            if (_options != null &&
+                _options.Value.DataAnnotationLocalizerProvider != null &&
+                _stringLocalizerFactory != null)
+            {
+                stringLocalizer = _options.Value.DataAnnotationLocalizerProvider(
+                    context.ModelMetadata.ContainerType ?? context.ModelMetadata.ModelType,
+                    _stringLocalizerFactory);
+            }
+
             foreach (var attribute in context.ValidatorMetadata.OfType<ValidationAttribute>())
             {
-                context.Validators.Add(new DataAnnotationsModelValidator(attribute));
+                if (stringLocalizer != null &&
```

Comment:
This logic seems odd here. If there's a `stringLocalizer` we should always pass it, the validator itself should determine which validation logic it should use (the validation attributes or the string localizer's).